src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/BoundedQueueTest.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+@file:Suppress("unused")
+
+package org.jetbrains.kotlinx.lincheck_test.representation
+
+import kotlinx.atomicfu.*
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.annotations.Param
+import org.jetbrains.kotlinx.lincheck.checkImpl
+import org.jetbrains.kotlinx.lincheck.paramgen.IntGen
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck_test.util.checkLincheckOutput
+import org.junit.Test
+
+/**
+ * This test is created to verify this [bug](https://github.com/JetBrains/lincheck/issues/209) is resolved.
+ * The problem was that we ran in infinite spin-lock in trace collection mode due to incorrect Loop detector work.
+ */
+@Param(name = "element", gen = IntGen::class, conf = "0:3")
+class BoundedQueueTest {
+    private val queue = QueueAdaptor<Int>()
+
+    @Operation
+    fun enqueue(@Param(name = "element") element: Int) = queue.enqueue(element)
+
+    @Operation
+    fun dequeue() = queue.dequeue() != null
+
+    @Test
+    fun modelCheckingTest() =
+        ModelCheckingOptions()
```

Comment:
Can you simplify the test configuration?

Reply:
I removed the setting and replaced it with a custom scenario configuration. I think it's a good approach to use in the test exactly the same scenario as caused the issue.